subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultFileSystemSnapshotter.java

```diff
@@ -248,6 +253,19 @@ private FileHashSnapshot fileSnapshot(File file, FileMetadataSnapshot fileDetail
         return new FileHashSnapshot(hasher.hash(file, fileDetails), fileDetails.getLastModified());
     }
 
+    private static final HashCode SIGNATURE = Hashing.md5().hashString(SymbolicLinkSnapshot.class.getName());
+    private SymbolicLinkHashSnapshot symbolicLinkSnapshot(FileTreeElement symbolicLinkDetails) {
+        try {
+            Hasher hasher = Hashing.md5().newHasher();
+            hasher.putHash(SIGNATURE);
+            hasher.putString(Files.readSymbolicLink(symbolicLinkDetails.getFile().toPath()).toFile().getPath());
+            hasher.putString(symbolicLinkDetails.getPath());
```

Comment:
I'm unsure if we should use the current `FileHasher` or create an `SymbolicLinkHasher`?

Reply:
I would leave this here until we find another place that needs it.

Reply:
The constant `SIGNATURE` was moved to `BrokenSymbolicLinkHashSnapshot` in the effort of making this piece of code better.